lib/rspec/expectations/failure_aggregator.rb

```diff
@@ -0,0 +1,159 @@
+module RSpec
+  module Expectations
+    # @private
+    class FailureAggregator
+      attr_reader :block_label, :metadata
+
+      def aggregate
+        RSpec::Support.with_failure_notifier(self) do
+          begin
+            yield
+          rescue ExpectationNotMetError => e
+            failures << e
+          rescue Exception => e
```

Comment:
hrmm, I actually don't understand yet how even the normal `ExpectationNotMetError` "resumes" after the first failure. Gonna go read the rspec-core PR, maybe there is something in there.